ts/sagas/startup/generateCryptoKeyPair.ts

```diff
@@ -0,0 +1,32 @@
+import { generate, deleteKey, CryptoError } from "io-react-native-crypto";
+import { call } from "typed-redux-saga/macro";
+import {
+  checkPublicKeyExists,
+  isKeyAlreadyGenerated,
+  setKeyAlreadyGenerated
+} from "../../utils/crypto";
+import { mixpanelTrack } from "./../../mixpanel";
+
+const KEY_NAME = "lp-temp-key";
+
+export function* generateCryptoKeyPair() {
+  try {
+    const keyExistsOnKeystore = yield* call(checkPublicKeyExists, KEY_NAME);
+    const keyAlreadyExists = yield* call(isKeyAlreadyGenerated, KEY_NAME);
+    const generateKey = !(keyAlreadyExists || keyExistsOnKeystore);
```

Comment:
Maybe a more explicit name could be used. eg. `isKeyExist` or `isKeyEmpty`

Reply:
Do you mean on line # 16 or the first two lines or all the lines? Usually if you use `is` you don't use `exists`.  Those are convention mutually exclusive. But I agree with you that we could improve the names of this flags. 

What do you think of the below proposal?

```ts
const keyAlreadyExistsOnKeystore = yield* call(checkPublicKeyExists, KEY_NAME);
const keyHasBeenAlreadyGenerated = yield* call(isKeyAlreadyGenerated, KEY_NAME);
const shouldWeGenerateKey = !(keyHasBeenAlreadyGenerated || keyAlreadyExistsOnKeystore);
```